src/assets.zig

```diff
@@ -294,29 +297,65 @@ fn registerRecipesFromZon(zon: ZonElement) void {
 
 pub const Palette = struct { // MARK: Palette
 	palette: main.List([]const u8),
-	pub fn init(allocator: NeverFailingAllocator, zon: ZonElement, firstElement: ?[]const u8) !*Palette {
+
+	pub fn init(allocator: NeverFailingAllocator, zon: ZonElement, firstElement: ?[]const u8) !*@This() {
```

Comment:
I generally do not like `@This()` because it makes it harder to read.
I think it should be only used in generics, where we don't have a dedicated name.